identity/aziot-identityd-config/src/lib.rs

```diff
@@ -124,31 +125,98 @@ pub enum ManualAuthMethod {
     X509 {
         identity_cert: String,
         identity_pk: String,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        csr_subject: Option<CsrSubject>,
     },
 }
 
-#[derive(Clone, Debug, serde::Deserialize, serde::Serialize)]
-#[serde(tag = "method")]
-#[serde(rename_all = "lowercase")]
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum CsrSubject {
+    CommonName(String),
+    #[serde(deserialize_with = "subject_from_key_value")]
```

Comment:
Do you need the `deserialize_with` ? With:

```rust
#[serde(rename = "CN")]
cn: String,

#[serde(flatten, skip_serializing_if = "BTreeMap::is_empty")]
rest: BTreeMap<String, String>,
```

... I get:

```
$ toml::from_str::<CsrSubject>("CN = \"foo\"\nDN = \"bar\"")

Ok(Subject { cn: "foo", rest: {"DN": "bar"} })

$ toml::from_str::<CsrSubject>("CN = \"foo\"")

Ok(Subject { cn: "foo", rest: {} })
```

Reply:
Not necessarily. We can always uppercase later in the code, but the uppercasing has to happen at some point since OpenSSL expects uppercased field identifiers. I can simplify the deserialization to use rename on the `cn` field and only uppercase the `BTreeMap` keys, though.

Reply:
>but the uppercasing has to happen at some point since OpenSSL expects uppercased field identifiers

Oh right, I forgot about that. Makes sense.